happi/audit.py

```diff
@@ -109,6 +109,42 @@ def check_args_kwargs_match(result: SearchResult) -> None:
            f'undefined variables found in document: {undefined}'
 
 
+def find_unfilled_mandatory_info(
+    result: SearchResult
+) -> list[str]:
+    """
+    Return all mandatory fields that are missing a value
+    """
+    return [info for info in result.item.mandatory_info
+            if getattr(result.item, info) is None]
+
+
+def find_unfilled_optional_info(
+    result: SearchResult
+) -> list[str]:
+    """
+    Return all optional fields that are missing a value
+    """
+    # cannot getattr for extraneous info, shortcircuit conditional
+    return [info for info in list(result.item.keys())
+            if info not in result.item.mandatory_info
+            and info not in result.item.extraneous.keys()
+            and getattr(result.item, info) is None]
+
+
+def check_unfilled_mandatory_info(result: SearchResult) -> None:
+    """
+    Check that all mandatory EntryInfo have a value.
+
+    This identifies problems originating from a mismatch between
+    an item's backend representation and desired container.
+    This may arise from manual editing of the backend database.
+    """
+    unfilled_info = find_unfilled_mandatory_info(result)
+    assert len(unfilled_info) == 0, \
```

Comment:
It is possible to run Python in such a way that it ignores all assert statements, so if this is used in try/except blocks it may be better to raise some other exception directly

Reply:
Oh TIL.  fixed in 38f77f3